### ISSUE TITLE
display custom states(events) for each day

### ISSUE THREAD

**Issue by marek-sed** [clauderic/react-infinite-calendar#6]
Hi,

Is there a way to display  states like birthday, federal holidays, events on each day?

**Comment by clauderic** [comment]
I think the best way to go about this would be to provide a custom day renderer function and a custom selected state renderer. Thoughts?

**Comment by marek-sed** [comment]
yes I was thinking something like that. Basically looking at date component at it is written now, 
it has 2 parts. Wrapper Component that handles click events and disabled state, and creates data needed for display component. And Display Component, which receives the needed data, and handles state('selected', 'normal'). So it would be nice if user can provide this custom display component.

What do you think?

I can implement, if you want I need this exact functionality in my app.

**Comment by clauderic** [comment]
I have a pretty good idea of how I would implement it, but I'm curious to see how you would implement it. How about you take a first stab at it and we'll take it from there :)

**Comment by marek-sed** [comment]
Yeah thats sounds like a good plan :)

**Comment by kennetpostigo** [comment]
@clauderic  @marek-sed it would be pretty neat if there was a tool-tip shown when an event is hovered on or clicked on. What do you guys think?

**Comment by jamestwiggle** [comment]
@marek-sed @clauderic what is the status on this?

**Comment by clauderic** [comment]
Haven't heard back from @marek-sed, I'll get around to tackling this next week. Tooltip would be nice for sure, but I'm probably going to build the logic for this to be as abstract as possible and then provide a couple of examples fecipes

**Comment by marek-sed** [comment]
Hi @clauderic  @jamestwiggle , sry I went silent, was quite busy. 
I just issued the pull request for this.

**Comment by shikelong** [comment]
when can you release the new npm package that merged this pull request ? 

Regards. @clauderic

**Comment by clauderic** [comment]
I'll take a look at it next week, I'm currently on vacation :)

**Comment by hiddenl00p** [comment]
@clauderic I understand this may be your side project, but do you know when you will be able to support this?

**Comment by pinder360** [comment]
@clauderic any update on this?

**Comment by oyeanuj** [comment]
@marek-sed @clauderic Also, curious if this is on any one of your radar still?

**Comment by clauderic** [comment]
This would be fairly simple to implement with the new architecture of `react-infinite-calendar@^2.0.0`.

This isn't something I have bandwidth to work on at the moment, but if you take a look at the implementation of the `withMultipleDates` HOC: https://github.com/clauderic/react-infinite-calendar/blob/master/src/Calendar/withMultipleDates.js. 

I think the implementation for displaying events would be fairly similar implementation.

**Comment by offsky** [comment]
I would be happy to take a stab at this, but Im new to recompose, so I could use a little help coming up to speed.  Can someone please explain what this line does from the above mentioned file?

`withProps(({displayDate, onSelect, setDisplayDate, scrollToDate, ...props}) => ({`

I think I understand that it is doing something magical to inject props into the secret wrapper container around Calendar. Im not quite sure where those 5 arguments come from and where they are going. In particular, where is "scrollToDate" defined? Thanks!

**Comment by arecvlohe** [comment]
@offsky He is using destructuring to select the props that are coming before it and modifying them in some way before passing them to the wrapped component. I hope that helps.

**Comment by awb99** [comment]
Is it possible to display events now? The pull request is there, but is it merged?

**Comment by daiky00** [comment]
@clauderic @Makashov is this already possible? please tell me is ready! it has been two years waiting on this feature
